File: 869-reordered-power-of-2/869-reordered-power-of-2.java
class Solution {
    
    private int[] count(int n) {
        int[] count = new int[10];
        while(n > 0) {
            count[n % 10]++;
            n /= 10;
        }
        return count;
    }
    
    public boolean reorderedPowerOf2(int n) {
        
        int[] digits = count(n);
        for(int i = 0; i < 31; i++)
            if(Arrays.equals(digits, count(1 << i)))
                return true;
        
        return false;
    }
}

File: 869-reordered-power-of-2/869-reordered-power-of-2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int n) {
        if(n == 0) return false;
        
        string str = "";
        while(n != 0) {
            str += to_string(n%10);
            n /= 10;
        }
        
        sort(str.begin(), str.end());
        
        int temp = 1;
        int power = pow(10, 9);
        
        while(temp <= power) {
            string t = to_string(temp);
            sort(t.begin(), t.end());
            if(str == t) return true;
            temp *= 2;
        }
        
        return false;
    }
};